guides/plugins/apps/starter/add-api-endpoint.md

```diff
@@ -0,0 +1,320 @@
+Slashes (or subdirectories) in the endpoint path are represented by a hyphen in the name of the directory that contains the script.
+
+```
+MyApiExtension/
+├─ Resources/
+│  ├─ scripts/
+│  │  ├─ store-api-swag-topseller/ <-- /store-api/script/swag/topseller
+│  │  │  ├─ topseller-script.twig
+├─ manifest.xml
+```
+
+### Add custom logic and install
+
+Let's start with a simple script to see it in action:
+
+{% code title="Resources/scripts/store-api-swag-topseller/topseller-script.twig" %}
+{% raw %}
+```twig
+{% block response %}
+    {% set response = services.response.json({ test: 'This is my API endpoint' }) %}
+    {% do hook.setResponse(response) %}
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+Next we will install the App using the Shopware CLI.
+
+Run this command from the root of the project directory.
+
+```shell
+shopware-cli project extension upload . --activate
+```
+
+This command will create a zip file from the specified extension directory (the one you are in), upload it to your configured store and activate it.
+
+### Call the endpoint
+
+You can call the endpoint using this curl command.
+
+{% hint style="info" %}
+Follow this guide for more information on using the Store API
+
+[**Store API Authentication & Authorization**](https://shopware.stoplight.io/docs/store-api/ZG9jOjEwODA3NjQx-authentication-and-authorisation)
+{% endhint %}
+
+```shell
+curl --request GET \
+  --url http://<your-store-url>/store-api/script/swag/topseller \
+  --header 'sw-access-key: insert-your-access-key'
+```
+
+which should return something like:
+
+```json
+{"apiAlias":"store_api_swag_topseller_response","test":"This is my API endpoint"}
+```
+
+However, instead of using curl we recommend using visual clients to test the API - such as [Postman](https://www.postman.com/downloads/) or [Insomnia](https://insomnia.rest/download).
+
+## Fill in the logic
+
+For now, our script is not really doing anything. Let's change that.
+
+{% code title="Resources/scripts/store-api-swag-topseller/topseller-script.twig" %}
+{% raw %}
+```twig
+{% block response %}
+
+    {% set categoryId = hook.request.categoryId %}
+
+    {% set criteria = {
+        aggregations: [
+            {
+                name: "categoryFilter",
+                type: "filter",
+                filter: [{
+                    type: "equals",
+                    field: "order.lineItems.product.categoryIds",
+                    value: categoryId
+                }],
+                aggregation: {
+                    name: "orderedProducts",
+                    type: "terms",
+                    field: "order.lineItems.productId",
+                    aggregation: {
+                        name: "quantityItemsOrdered",
+                        type : "sum",
+                        field: "order.lineItems.quantity"
+                    }
+                }
+            }
+        ]
+    } %}
+
+    {% set orderResult = services.repository.search('order', criteria) %}
```

Comment:
```suggestion
    {% set orderAggregations = services.repository.aggregate('order', criteria) %}
```

if you are only interested in the aggregations you should use the `aggregate` method directly, instead of `search` because `search` would also fetch all orders which is quite slow and resource intensive